lib/bambora/errors.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Bambora
+  # All library-specific errors inherit from this error class. This helps make it easy for integrations to rescue
+  # specifically from errors raised by this library.
+  class Error < StandardError
+    attr_reader :payload
```

Comment:
Developing custom error classes allows us to provide a better developer experience around API errors.

With these error classes, we can parse the API payload ourselves in the library, and raise an appropriate error that the developer can rescue from. This will reduce developer effort in integrating with this library, and allow them to create clearer code.

```ruby
# BEFORE

response = Bambora::V1::Reports::Merchants.get_all(credentials: credentials)

if response[:message] == 'Authentication failed'
  # Handle authentication failure.
elsif response[:Message] == 'The request is invalid.'
  # Handle invalid request. Note that the uppercase M is not a typo.
else
  # Handle merchants.
end

# AFTER

begin
  merchants = Bambora::V1::Reports::Merchants.get_all(credentials: credentials)

  # Handle merchants.
rescue Bambora::InvalidAuthenticationError
  # Handle authentication failure.
rescue Bambora::InvalidRequestError
  # Handle invalid request.
rescue Bambora::Error
  # Handle all other errors.
end
```

Here, we create a base `Error` class, which all errors are meant to inherit from. This easily allows a developer to isolate and rescue from all (and only) errors raised by this library. We provide read access to the `#payload` in case the developer wants to use it.